server/decoders/systems/index.ts

```diff
@@ -0,0 +1,26 @@
+// ---------------------------------------------------------------------------------------------------------------------
+// System Specific Decoders
+// ---------------------------------------------------------------------------------------------------------------------
+
+import { Decoder, fail } from 'decoders';
+
+// System Decoders
+import { wfrpSysDetailsDecoder } from './wfrp';
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+export function sysDetailsDecoder<T extends Record<string, unknown>>(system : string) : Decoder<T>
+{
+    // TODO: Add more systems!
+
+    // Add an entry for every possible system type, and return the decoder for those system details.
+    switch (system)
+    {
+        case 'wfrp':
+            return wfrpSysDetailsDecoder as Decoder<T>;
+        default:
+            return fail(`Unknown system '${ system }'.`);
+    } // end switch
+} // end sysDetailsDecoder
+
+// ---------------------------------------------------------------------------------------------------------------------
```

Comment:
I'm not sure the purpose of this file. Granted, it's being viewed in isolation, but it looks like a kludge to support a specific system, and would require updating as new systems are added. In short, maybe some comment at the top indicating the intended purpose / why of this?

Reply:
You got to the heart of one of the minor annoyances I have, and this is the fix.

1. Yes, we need to add the rest of the systems here.
2. It's a bit of a kludged; but this isn't done. I've got more systems done locally, atm, and will have all of them before I take this out of draft status.

As for it's purpose? I need to be able to pass a string to it, and have it pass me back the right sub-system decoder. Or, I move this logic into both places who use this, and import all possible system decoders into the character model; doesn't exactly improve things, just moves which file has this problem.

Turns out, I need a run-time check of the system key on a character to know the right one to use; I can't base it off type, since types are compiled out at runtime; Comparing or working with Typescript Types is only possible when they're full classes, and even then, very rudimentary and awkward.

Reply:
So the core is given a character we need to be able to decode the system the character uses based on the 'system' property in the character, which is a string. Do I have that right?

Reply:
yup.